flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/rules/FlinkStreamRuleSets.scala

```diff
@@ -72,10 +72,15 @@ object FlinkStreamRuleSets {
 
   /** RuleSet to reduce expressions */
   private val REDUCE_EXPRESSION_RULES: RuleSet = RuleSets.ofList(
-    CoreRules.FILTER_REDUCE_EXPRESSIONS,
     CoreRules.PROJECT_REDUCE_EXPRESSIONS,
+    CoreRules.FILTER_REDUCE_EXPRESSIONS,
     CoreRules.CALC_REDUCE_EXPRESSIONS,
-    CoreRules.JOIN_REDUCE_EXPRESSIONS
+    CoreRules.WINDOW_REDUCE_EXPRESSIONS,
```

Comment:
I wonder is there a way to put these rules in Java rather than Scala - as I believe the direction is to remove Scala from the table planner.